components/availability/src/Availability.svelte

```diff
@@ -63,17 +60,49 @@
       .slice(0, -1) // dont show the 25th hour
       .map((time) => {
         const endTime = timeMinute.offset(time, slot_size);
-
-        let slotIsAvailable = true; // default
-        if (available_times.length) {
-          slotIsAvailable = available_times.some((slot) => {
-            return time >= slot.start_time && endTime <= slot.end_time;
+        let freeCalendars = [];
+
+        let availability = "available"; // default
+
+        if (calendars.length) {
+          calendars.forEach((c) => {
+            if (c.availability === "busy") {
+              if (
+                !c.timeslots.some(
+                  (slot) => time >= slot.start_time && endTime <= slot.end_time,
+                )
+              ) {
+                freeCalendars.push(c.email_address);
+              }
+            } else if (c.availability === "free" || !c.availability) {
+              // if they pass in a calendar, but don't have availability, assume the timeslots are available.
+              if (
+                c.timeslots.some(
+                  (slot) => time >= slot.start_time && endTime <= slot.end_time,
+                )
+              ) {
+                freeCalendars.push(c.email_address);
+              }
+            }
           });
         }
 
+        if (calendars.length) {
+          if (freeCalendars.length) {
+            if (freeCalendars.length === calendars.length) {
+              availability = "free";
+            } else {
+              availability = "partial";
```

Comment:
Can a calendar's availability be partial? If it can, the typing should be updated.

Reply:
It can't -- a timeslot can be "partial", indicating the multiple parties involved are not fully free but not fully busy, but an individual person's calendar cannot be partial.